Adding volume to lib_openshift.

kwoodson opened the pull request.

kwoodson commented:
aos-ci-test

ashcrow commented on roles/lib_openshift/src/doc/volume:
```diff
@@ -0,0 +1,98 @@
+    required: false
+    default: /etc/origin/master/admin.kubeconfig
+    aliases: []
+  debug:
```
I didn't see this in the class.

kwoodson replied:
@ashcrow, Its a little difficult to follow but I believe it is here.
https://github.com/kwoodson/openshift-ansible/blob/oc_volume/roles/lib_openshift/src/ansible/oc_volume.py#L14
Then passed into the class here.
https://github.com/kwoodson/openshift-ansible/blob/oc_volume/roles/lib_openshift/src/class/oc_volume.py#L116

ashcrow commented on roles/lib_openshift/src/doc/volume:
```diff
@@ -0,0 +1,98 @@
+    required: false
+    default: None
+    aliases: []
+author:
```
Looks like resource_name, vol_name, and verbose are missing.

kwoodson commented:
@ewolinetz, please take a look at the configmap code added.

@ashcrow, if you get a few minutes.  I updated this module with the bits to add configmap to volumes.

ashcrow commented on roles/lib_openshift/library/oc_volume.py:
```diff
@@ -1750,24 +1756,28 @@ def create_volume_structure(volume_info):
             volume_mount = {'mountPath': volume_info['path'],
                             'name': volume_info['name']}
-        elif volume_info['type'] == 'pvc':
+        elif volume_type == 'pvc' or volume_type == 'persistentvolumeclaim':
```
:+1:

ashcrow commented on roles/lib_openshift/library/oc_volume.py:
```diff
@@ -1750,24 +1756,28 @@ def create_volume_structure(volume_info):
         volume_mount = None
         volume = {'name': volume_info['name']}
-        if volume_info['type'] == 'secret':
+        volume_type = volume_info['type'].lower()
```
:+1: I think switching back to case insensitive is best.

kwoodson replied:
Yeah, this was my fault.  @mtnbikenc had made changes to this file to do the insensitive case checking.  I stomped his changes in the first PR so I reverted back.  That was my fault.

kwoodson commented:
@ewolinetz, the failures mentioned by @ashcrow were in the new test I added for configmap.

I have fixed the test and ran `tox` locally and everything looks good.  Let's try this again!

ashcrow commented:
aos-ci-test
[test]

mtnbikenc commented:
@ashcrow I've not had success with putting aos-ci-test in the same comment with `bracket-test-bracket`.  So I've just done each in their own comment.

mtnbikenc commented:
aos-ci-test

kwoodson commented:
Seems like this flaked out on:
>                 Message: "the server has asked for the client to provide credentials (get deploymentConfigs history-limit)",

Rerunning.

kwoodson commented:
aos-ci-test

kwoodson commented:
[test]

kwoodson commented:
@stevekuznetsov, @sdodson, any ideas on this one?  

I think its unrelated to oc_volume since this is brand new code that is unreferenced.

stevekuznetsov commented:
@smarterclayton does this look like namespace cleanup failures in serial tests?
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install/53/#showFailuresLink

e2e framework bug?

stevekuznetsov commented:
That is one noisy bot ....

sdodson commented:
[test]

kwoodson commented:
[merge]

smarterclayton commented:
@stevekuznetsov yes, if more than one serial test fails it's almost certainly because of namespace cleanup.
